apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.IntegrationTests/Trace/TraceTest.cs

```diff
@@ -0,0 +1,420 @@
+            var startTime = Timestamp.FromDateTime(DateTime.UtcNow);
+
+            var builder = new WebHostBuilder().UseStartup<TraceTestNoBufferApplication>();
+            using (TestServer server = new TestServer(builder))
+            {
+                var client = server.CreateClient();
+
+                // The first request is not traced, sleep long enough 
+                // for the rate limiter to allow a request, then the second
+                // request is traced.
+                await client.GetAsync($"/Trace/TraceLabels/{testId}");
+                Thread.Sleep(TimeSpan.FromSeconds(2));
+                await client.GetAsync($"/Trace/Trace/{testId}");                
+            }
+
+            var spanNameTrace = TraceController.GetMessage(nameof(TraceController.Trace), testId);
+            var spanNameLabels = TraceController.GetMessage(nameof(TraceController.TraceLabels), testId);
+            Assert.Null(await GetTrace(spanNameLabels, startTime, expectTrace: false));
+            Assert.NotNull(await GetTrace(spanNameTrace, startTime));
+        }
+
+        [Fact]
+        public async Task Trace_Buffer()
+        {
+            string testId = Utils.GetTestId();
+            var startTime = Timestamp.FromDateTime(DateTime.UtcNow);
+
+            var builder = new WebHostBuilder().UseStartup<TraceTestBufferHighQpsApplication>();
+            using (TestServer server = new TestServer(builder))
+            {
+                var client = server.CreateClient();
+
+                // Make a trace with a small span that will not cause the buffer to flush.
+                await client.GetAsync($"/Trace/Trace/{testId}");
+                var spanName = TraceController.GetMessage(nameof(TraceController.Trace), testId);
+                Assert.Null(await GetTrace(spanName, startTime, expectTrace: false));
+
+                // Make a large trace that will flush the buffer.
+                await client.GetAsync($"/Trace/TraceStackTrace/{testId}");
+            }
+
+            var spanNameTrace = TraceController.GetMessage(nameof(TraceController.Trace), testId);
+            var spanNameStack = TraceController.GetMessage(nameof(TraceController.TraceStackTrace), testId);
+            Assert.NotNull(await GetTrace(spanNameTrace, startTime));
+            Assert.NotNull(await GetTrace(spanNameStack, startTime));
+        }
+
+        [Fact]
+        public async Task Trace_Exception()
+        {
+            string testId = Utils.GetTestId();
+            var startTime = Timestamp.FromDateTime(DateTime.UtcNow);
+
+            var builder = new WebHostBuilder().UseStartup<TraceTestNoBufferHighQpsApplication>();
+            using (TestServer server = new TestServer(builder))
+            {
+                var client = server.CreateClient();
+                try
+                {
+                    await client.GetAsync($"/Trace/ThrowException/{testId}");
+                }
+                catch (Exception e)
```

Comment:
The variable name `e` is not necessary here (neither is the type name actually...this can just be `catch`).